js/stores/helpers/boundary-utils.ts

```diff
@@ -82,35 +82,86 @@ export function unhighlightBoundary(field: FieldStore) {
+  if (polarCapField) {
     // latitudinal boundary
-    const capPlateId = polarCapPlate.id;
-    const nonCapPlateId = plate === polarCapPlate ? otherPlate.id : plate.id;
-    plates = orientation === "northern-latitudinal"
-              ? [capPlateId, nonCapPlateId]
-              : [nonCapPlateId, capPlateId];
+    const nonCapField = field === polarCapField ? otherField : field;
+    fields = orientation === "northern-latitudinal"
+              ? [polarCapField, nonCapField]
+              : [nonCapField, polarCapField];
   } else {
     // longitudinal boundary
-    const platePos = toSpherical(plate.center);
-    const otherPlatePos = toSpherical(otherPlate.center);
-    plates = platePos.lon < otherPlatePos.lon
-              ? [otherPlate.id, plate.id]
-              : [plate.id, otherPlate.id];
+    const platePos = toSpherical(field.plate.center);
+    const otherPlatePos = toSpherical(otherField.plate.center);
+    fields = platePos.lon < otherPlatePos.lon
+              ? [otherField, field]
+              : [field, otherField];
+  }
+  return { orientation, fields };
+}
+
+export function convertBoundaryTypeToHotSpots(boundaryInfo: IBoundaryInfo): IHotSpot[] {
+  const [field0, field1] = boundaryInfo.fields || [];
+  const type = boundaryInfo.type;
+  switch (boundaryInfo.orientation) {
+  case "northern-latitudinal":
+    if (field0) {
+      const latLon = toSpherical(field0.absolutePos);
+      const lat = latLon.lat + 0.15;
+      const lon = latLon.lon;
+      const newPos = toCartesian([lat, lon]);
+      const rotationAxis = newPos.clone().normalize();
+      const trueNorth = getNorthVector(lat, lon);
+      const force = type === "convergent" ? trueNorth.applyAxisAngle(rotationAxis, Math.PI) : trueNorth;
+      return [{ position: newPos, force }];
+    }
+    break;
+  case "longitudinal":
+    if (field0 && field1) {
+      const latLon0 = toSpherical(field0.absolutePos);
+      const lat0 = latLon0.lat;
+      // TODO: it should be adjusted for extreme lat values in a better way
+      const kLonDiff = 0.2 + Math.pow(Math.abs(lat0), 2) * 0.2;
+      const lon0 = latLon0.lon - kLonDiff;
+      const latLon1 = toSpherical(field1.absolutePos);
+      const lat1 = latLon1.lat;
+      const lon1 = latLon1.lon + kLonDiff;
+      const latAvg = 0.5 * (lat0 + lat1);
+      const newPos0 = toCartesian([latAvg, lon0]);
+      const newPos1 = toCartesian([latAvg, lon1]);
+      const rotationAxis0 = newPos0.clone().normalize();
+      const rotationAxis1 = newPos1.clone().normalize();
+      const trueNorth0 = getNorthVector(latAvg, lon0);
+      const trueNorth1 = getNorthVector(latAvg, lon1);
+      const force0 = type === "convergent" ? trueNorth0.applyAxisAngle(rotationAxis0, 0.5 * Math.PI) : trueNorth0.applyAxisAngle(rotationAxis0, -0.5 * Math.PI);
+      const force1 = type === "convergent" ? trueNorth1.applyAxisAngle(rotationAxis1, -0.5 * Math.PI) : trueNorth1.applyAxisAngle(rotationAxis1, 0.5 * Math.PI);
```

Comment:
I just noticed you could DRY up these lines a bit by moving the ternary to the rotation angle argument.

Reply:
Thanks, I just did it, so I'll push it as a part of another PR.